stdlib/public/core/Stride.swift

```diff
@@ -134,15 +134,63 @@ public protocol Strideable: Comparable {
   /// If this type's `Stride` type conforms to `BinaryInteger`, then for a
   /// value `x`, a distance `n`, and a value `y = x.advanced(by: n)`,
   /// `x.distance(to: y) == n`. Using this method with types that have a
-  /// noninteger `Stride` may result in an approximation.
+  /// noninteger `Stride` may result in an approximation. If the result of
+  /// advancing by `n` is not representable as a value of this type, then a
+  /// runtime error may occur.
```

Comment:
cc @natecook1000 for this documentation change.

Reply:
The doc change looks fine, but isn't this what this change is meant to avoid? Or are there still cases where advancing past e.g. `Int8.max` traps?

Reply:
This PR is about `stride` working around this limitation of `advanced(by:)`. There is no change to the behavior of `advanced(by:)`, which must return a value and therefore must fail to return if a value is not representable.

Reply:
Would so nice if `advanced(by:)` would be failable.

Reply:
Got it! I misunderstood what I was looking at.